lib/metasploit/framework/login_scanner/http.rb

```diff
@@ -294,7 +294,7 @@ def attempt_login(credential)
 
           begin
             response = send_request('credential'=>credential, 'uri'=>uri, 'method'=>method)
-            if response && response.code == 200
+            if response && (response.code == 200 || (response.code >= 300 && response.code < 400))
```

Comment:
This logic seems application specific to me; I'm not sure we can make the assumption that a 3xx is always a sign of a successful login

Reply:
I'm a little hesitant w/ the >= 300 as well, while *many* apps do use a 301 to redirect to say `/admin`, some also redirect back to login screen.  Maybe set this as an advanced option so it can be configured?

Reply:
There is a check for `res.redirect?` I would be happy to use instead of `>=` just needed to start the conversation here.

As to an advanced option that might be a path, something like `RedirectsOnSuccess` with a default as `false` that might provide a reasonable compromise.

Reply:
The complex part here is that this is a login_scanner class so `datastore` access should be avoided requiring some sort of injection option maybe a new accessor or some optional `block` to testing success... not sure what pattern I like, will see about adding something here that feels usable for user and developer accessing this class.

Reply:
An idea updated here, let me know if this makes sense.